### ISSUE TITLE
Custom device statuses fail to unmarshal, causing entire call to fail

### ISSUE THREAD

**Issue by TomFletcher0** [netbox-community/go-netbox#179]
We have a custom device status, "Racked" set up using the normal method in Netbox (no code change required in Netbox), but then when I search (`client.DcimAPI.DcimDevicesList(ctx)`) for devices (with or without filters) the client throws error:

`DcimDevicesList failed, racked is not a valid DeviceStatusValue`

My workaround is to add the custom statuses to model_device_status_label.go & model_device_status_value.go, but this requires a fork and change to generated code.

It would be good even if it failed to unmarshal those particular devices with custom statuses, at least the rest of the devices from the search (those with supported statuses) could be returned.

**Comment by ljb2of3** [comment]
I've also run into this. I have multiple custom statuses. This seems like a systemic problem since netbox allows custom choices for many different fields so this could happen with any number of fields, not just device status.

**Comment by faildruid** [comment]
Also experiencing this issue.  It appears that the library cannot read values defined in  [FIELD_CHOICES](https://netboxlabs.com/docs/netbox/en/stable/configuration/data-validation/#field_choices) config value.

**Comment by nutgood** [comment]
Indeed, since those are not API defined types and the schema hardcodes possible values, it's impossible for us to implement a way to fix this issue.

In the meantime, you shouldn't have to modify the source, instead, you can simply do this (I think):

```go

netbox.AllowedDeviceStatusLabelEnumValues = append(netbox.AllowedDeviceStatusLabelEnumValues, "racked")
netbox.AllowedDeviceStatusValueEnumValues = append(netbox.AllowedDeviceStatusValueEnumValues, "racker")
client := netbox.NewApiClientFor(...)

```

Closing as won't fix since there's nothing (else) I can do.
